index.d.ts

```diff
@@ -0,0 +1,111 @@
+type Distance = {
+  distanceX: number;
+  distanceY: number;
+};
+
+interface KeyEffect {
+  initialX: number;
+  initialY: number;
+  name?: string;
+  description?: string;
+}
+
+interface KeyMap<E> {
+  keys: Record<string, E>;
+  name?: string;
+  description?: string;
+}
+
+interface KeyMappingConfig {
+  dpad?: KeyMap<KeyEffect & Distance>[];
+  tap?: KeyMap<KeyEffect>[];
+  swipe?: KeyMap<KeyEffect & Distance>[];
+}
+
+type DeviceRendererKeyMapping = {
+  enable(isEnable: boolean): void;
+  setConfig(config: KeyMappingConfig): void;
+  activeKeyMappingDebug(isTraceActivate: boolean, isGridActivate: boolean): void;
+};
+
+type VmEvent = 'beforeunload' | 'fingerprint' | 'gps' | 'BATTERY_LEVEL'
+
+type VmCommunication = {
+  disconnect(): void;
+  addEventListener(event: VmEvent, callback: (msg: unknown) => void): void;
+  sendData(data: { channel: string; messages: string[] });
+};
+
+type Utils = {
+  getRegisteredFunctions(): unknown[];
+};
+
+type Media = {
+  mute(): void;
+  unmute(): void;
+};
+
+type Template = 'bootstrap'
+  | 'fullscreen'
+  | 'fullwindow'
+  | 'renderer'
+  | 'renderer_minimal'
+  | 'renderer_no_toolbar'
+  | 'renderer_partial';
+
+interface Options {
```

Comment:
small question on the Options type: will we be able to override it from the Shadow repo? There are a few more options that are set in the Shadow player compared to the Geny player.